### PR TITLE
Inserting memory_limit parameter within LocalCluster docstring

### PR DESCRIPTION
**Pull request by crislanarafael**
Insert memory_limit parameter into the LocalCluster docstring and adding notes that the argument:
- Sets memory limit per worker
- Cannot exceed the total memory available

Closes #8224

- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

Admins can comment `ok to test` to allow this one PR to run or `add to allowlist` to allow all future PRs from the same author to run.

**Comment by crislanarafael** [comment]
Addresses issue #8224

File: distributed/deploy/local.py [hunk 1]
```diff
@@ -30,6 +30,8 @@ class LocalCluster(SpecCluster):
     ----------
     n_workers: int
         Number of workers to start
+    memory_limit: str
```
Comment (gjoseph92) [inline distributed/deploy/local.py:33]:
```suggestion
    memory_limit: str, float, int, or None, default "auto"
```
Here you can see all the many confusing possibilities for the `memory_limit` parameter: https://github.com/dask/distributed/blob/caf51895f63de4338a5a0c7862ecba0b8f7eb002/distributed/worker_memory.py#L383-L405

We should mention these options in the docstring, maybe like
```
* If None or 0, no limit is applied.
* If "auto", the total system memory is split evenly between the workers.
* If a float, that fraction of the system memory is used *per worker*.
* If a string giving a number of bytes (like ``"1GiB"``), that amount is used *per worker*.
* If an int, that number of bytes is used *per worker*.
```

File: distributed/deploy/local.py [hunk 2]
```diff
@@ -30,6 +30,8 @@ class LocalCluster(SpecCluster):
...
+        Sets the memory limit per worker. Cannot exceed the total memory available.
```
Comment (gjoseph92) [inline distributed/deploy/local.py:34]:
```suggestion
        Sets the memory limit *per worker*.
        
        Note that the limit will only be enforced when ``processes=True``, and the limit is only
        enforced on a best-effort basis—it's still possible for workers to exceed this limit.
```
Technically, exceeding the total memory available is okay: https://github.com/dask/distributed/blob/caf51895f63de4338a5a0c7862ecba0b8f7eb002/distributed/worker_memory.py#L404

And there are some issues with the implementation of the memory limit https://github.com/dask/distributed/issues/6177, so I'd also like to document that.

**Review by gjoseph92 — COMMENTED** [review]
Thanks for the fix @crislanarafael, this will be helpful for users to see. Since this is a funky parameter there are a couple other details worth mentioning.

**Comment by ncclementi** [comment]
add to allowlist

**Review by jrbourbeau — COMMENTED** [review]
Just checking in here, @crislanarafael are you okay to address @gjoseph92's review comments, or do you have any additional questions?

File: distributed/deploy/local.py (hunk 1 above)
Comment (crislanarafael) [reply distributed/deploy/local.py:33]: Oh that's good to know!

**Comment by crislanarafael** [comment]
These are good suggestions and I'll be sure to address these in the docstring for memory_limit

**Review by jrbourbeau — COMMENTED** [review]
Thanks @crislanarafael! It looks like there are some code linting errors that's causing a [CI build to fail](https://github.com/dask/distributed/runs/7846455932?check_suite_focus=true). For reference, we have a set of pre-commits to run these types of linting checks locally (see this documentation page for more details
https://docs.dask.org/en/stable/develop.html#code-formatting)

**Comment by crislanarafael** [comment]
I managed to fix the linting/pre-commit hooks errors after double checking the error messages. Let me know if there's anything else!

File: distributed/deploy/local.py [hunk 3]
```diff
@@ -100,6 +110,7 @@ class LocalCluster(SpecCluster):
     def __init__(
         self,
         name=None,
+        memory_limit=None,
```
Comment (gjoseph92) [inline distributed/deploy/local.py:113]: I don't think we want to add this parameter to `LocalCluster`. The variable `memory_limit` isn't used anywhere else in the body of this function. So by adding it here, it's "sucking up" the `memory_limit` argument and not passing it on to the worker, so in the end, the memory limit doesn't actually get applied.

Previously, when you called `LocalCluster(..., memory_limit=...)`, the `memory_limit` argument would end up in `**worker_kwargs`. That kwargs dict would end up here: https://github.com/dask/distributed/blob/3647cfefd004281dac0ffa92349e23bc2a94d68a/distributed/deploy/local.py#L235

Through a chain of calls, those `worker_kwargs`, including `memory_limit`, would eventually be passed into the `Worker` class. The `LocalCluster` doesn't actually use the argument directly.

You can see how some tests are failing indicating that the `memory_limit` wasn't passed down: https://github.com/dask/distributed/runs/7846945041?check_suite_focus=true#step:11:1915

Comment (crislanarafael) [reply distributed/deploy/local.py]: Thank you for the explanation!

**Comment by ian-r-rose** [comment]
Is there anything more to be done here, or is this ready to merge?

**Review by gjoseph92 — APPROVED** [review]
I think this is good—thanks @crislanarafael!

**Comment by ian-r-rose** [comment]
Thanks for the review  @gjoseph92 !

**Comment by crislanarafael** [comment]
Thanks @gjoseph92, glad to see my first merge on here!

**Pull request merged**